src/main/java/io/netty/incubator/codec/quic/QuicheSendInfo.java

```diff
@@ -83,33 +90,45 @@ private static long readLen(long address) {
      * @param memory the memory address of {@code quiche_send_info}.
      * @param address the {@link InetSocketAddress} to write into {@code quiche_send_info}.
      */
-    static void write(long memory, InetSocketAddress address) {
-        long sockaddr = sockAddress(memory);
-        int len = SockaddrIn.write(sockaddr, address);
-        switch (Quiche.SIZEOF_SOCKLEN_T) {
-            case 1:
-                PlatformDependent.putByte(memory + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, (byte) len);
-                break;
-            case 2:
-                PlatformDependent.putShort(memory + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, (short) len);
-                break;
-            case 4:
-                PlatformDependent.putInt(memory + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, len);
-                break;
-            case 8:
-                PlatformDependent.putLong(memory + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, len);
-                break;
-            default:
-                throw new IllegalStateException();
+    static void setSendInfo(ByteBuffer memory, InetSocketAddress address) {
+        int position = memory.position();
+        int sockaddrIdx = sockAddressIdx(position);
+        try {
+            memory.position(sockaddrIdx);
+            int len = SockaddrIn.setAddress(memory, address);
+            switch (Quiche.SIZEOF_SOCKLEN_T) {
+                case 1:
+                    memory.put(position + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, (byte) len);
+                    break;
+                case 2:
+                    memory.putShort(position + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, (short) len);
+                    break;
+                case 4:
+                    memory.putInt(position + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, len);
+                    break;
+                case 8:
+                    memory.putLong(position + Quiche.QUICHE_SEND_INFO_OFFSETOF_TO_LEN, len);
+                    break;
+                default:
+                    throw new IllegalStateException();
+            }
+        } finally {
+            memory.position(position);
         }
     }
 
+    static boolean isSameAddress(ByteBuffer memory, ByteBuffer memory2) {
```

Comment:
Looks like the same method is defined in `QuicheRecv Info` too

Reply:
it is but it uses different offsets into the memory